front/lib/api/assistant_builder/server_side_helpers.ts

```diff
@@ -0,0 +1,166 @@
+import type {
+  AgentConfigurationType,
+  AppType,
+  CoreAPITable,
+  DataSourceType,
+} from "@dust-tt/types";
+import {
+  ConnectorsAPI,
+  CoreAPI,
+  isDustAppRunConfiguration,
+  isRetrievalConfiguration,
+  isTablesQueryConfiguration,
+} from "@dust-tt/types";
+
+import type { AssistantBuilderInitialState } from "@app/lib/api/assistant_builder/types";
+import { getDataSource } from "@app/lib/api/data_sources";
+import type { Authenticator } from "@app/lib/auth";
+import { tableKey } from "@app/lib/client/tables_query";
+import logger from "@app/logger/logger";
+
+export async function buildInitialState({
```

Comment:
had to move that stuff here for nextJS bundling reasons -- it kept trying to import it into the client, and some of the imports here don't work in the browser.